pynest/nest/tests/test_connect_arrays.py

```diff
@@ -20,29 +20,52 @@
 # along with NEST.  If not, see <http://www.gnu.org/licenses/>.
 
 import unittest
-import nest
 import numpy as np
 
+import nest
+from nest.lib.hl_api_exceptions import NESTErrors
+
 nest.set_verbosity('M_WARNING')
 
 
+non_unique = np.array([1, 1, 3, 5, 4, 5, 9, 7, 2, 8], dtype=np.uint64)
+
+
 class TestConnectArrays(unittest.TestCase):
 
     def setUp(self):
         nest.ResetKernel()
 
-    def test_connect_arrays(self):
+    def test_connect_arrays_unique(self):
         """Connecting NumPy arrays of node IDs"""
         n = 10
         nest.Create('iaf_psc_alpha', n)
         sources = np.arange(1, n+1, dtype=np.uint64)
         targets = np.arange(1, n+1, dtype=np.uint64)
-        weights = np.ones(len(sources))
-        delays = np.ones(len(sources))
-        syn_model = 'static_synapse'
+        weights = 1.5
+        delays = 1.4
+
+        nest.Connect(sources, targets, syn_spec={'weight': weights, 'delay': delays})
+
+        conns = nest.GetConnections()
+
+        self.assertEqual(len(conns), n*n)
+
+        for c in conns:
+            self.assertTrue(np.isclose(c.weight, weights))
+            self.assertTrue(np.isclose(c.delay, delays))
```

Comment:
To avoid the unhelpful error message `False is not True` in event of failure, don't use `self.assertTrue()`. One alternative is to use functions from [`numpy.testing`](https://numpy.org/doc/stable/reference/routines.testing.html) to assert the results.